internal/controller/providers/pxc/provider.go

```diff
@@ -266,6 +231,40 @@ func (p *Provider) Status(ctx context.Context) (everestv1alpha1.DatabaseClusterS
 	return status, nil
 }
 
+// when a PXC restore is in progress, we will retry reconciliation
+// after the specified duration.
+const defaultRestoreRequeueDuration = 15 * time.Second
+
+func (p *Provider) isRestoreInProgress(ctx context.Context) (bool, error) {
+	var pxcRestoreList pxcv1.PerconaXtraDBClusterRestoreList
+	if err := p.C.List(ctx, &pxcRestoreList, client.InNamespace(p.DB.GetNamespace())); err != nil {
+		return false, err
+	}
+	for _, restore := range pxcRestoreList.Items {
+		if restore.Status.State == pxcv1.RestoreFailed || restore.Status.State == pxcv1.RestoreSucceeded || restore.Spec.PXCCluster != p.DB.GetName() {
```

Comment:
I'm wondering if we could check the `dbr` resources instead of the `pxc-restore` - for `dbr` we have this `IsComplete()` which would avoid duplicated logic

Reply:
updated